tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/AspNetCore/BlockingMiddleware.cs

```diff
@@ -0,0 +1,116 @@
+﻿// <copyright file="BlockingMiddleware.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+#if !NETFRAMEWORK
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Transports;
+using Microsoft.AspNetCore.Http;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.AspNetCore;
+
+internal class BlockingMiddleware
+{
+    private readonly bool _runSecurity;
+    private readonly bool _endPipeline;
+    private readonly RequestDelegate? _next;
+
+    internal BlockingMiddleware(RequestDelegate? next = null, bool runSecurity = false, bool endPipeline = false)
+    {
+        _next = next;
+        _runSecurity = runSecurity;
+        _endPipeline = endPipeline;
+    }
+
+    private static Task WriteResponse(HttpContext context, SecuritySettings settings, out bool endedResponse)
+    {
+        var httpResponse = context.Response;
+        if (!httpResponse.HasStarted)
+        {
+            httpResponse.Clear();
+            foreach (var cookie in context.Request.Cookies)
+            {
+                httpResponse.Cookies.Delete(cookie.Key);
+            }
+
+            httpResponse.Headers.Clear();
+            httpResponse.StatusCode = 403;
+            var template = settings.BlockedJsonTemplate;
+            if (context.Request.Headers["Accept"].ToString().Contains("text/html"))
```

Comment:
Just FYI, in .NET 6/7 `IHeaderDictionary` has an `Accept` property, which is more performant. Not saying you should definitely accept this extra complexity, just an FYI 🙂 

You may also want to consider not doing a ToString(), and instead iterating over the StringValues, as if they send multiple headers, this does a `string.Concat()` internally.

```suggestion
#if NET6_0_OR_GREATER
            if (context.Request.Headers.Accept.ToString().Contains("text/html"))
#else
            if (context.Request.Headers["Accept"].ToString().Contains("text/html"))
#endif
```

Reply:
looping through headers in 60213612bd9dc48fe4f254b95d896f2a95e759d4 
I ll add a small note in our backlog for the `Accept `property thank you!